blog/content/post/2019-10-24-meta-pcengines-xen.md

```diff
@@ -0,0 +1,128 @@
+---
+# What is the Xen Project?
+
+Xen Project creates a software system that allows the execution of multiple virtual guest operating systems simultaneously on a single physical machine.
+It is valuable when you want to create a complex security software system.
+None of the guest operating systems have direct access to the host hardware.
+
+A Xen hypervisor manages the virtual machine guests known as domains.
+It is controlled by a special domain called domain0 (dom0). Only the hypervisor
+has access to the physical outputs of the platform. Depending on the guest type
+(HVM or PV) hypervisor allows unprivileged domains (domUs) to use virtual
+hardware devices or passes the links to the physical hardware. To know more look
+at the Xen Project [wiki site](https://wiki.xenproject.org/wiki/Xen_Project_Beginners_Guide).
+
+<i>
+    <h2><p align=center>Xen Project architecture </p>
+</i>
+
+![Diagram](https://cloud.3mdeb.com/index.php/s/HNmc8yqbcCQzCje/preview)
+
+# How to build the dom0 on the PC Engines APU2?
+
+For this purpose, we will use the [meta-pcengines repository](https://github.com/pcengines/meta-pcengines.git).
+This is the yocto layer, which allows creating a minimal image of the dom0.
+The meta-pcengines needs a kas tool to construct the dom0 build.
+Installation and configuration of that tool were described in detail in the
+previous [blog](https://blog.3mdeb.com/2019/2019-02-07-kas/).
+
+Once you have got the kas, clone the [meta-pcengines repository](https://github.com/pcengines/meta-pcengines.git).
+In the main directory, you can see the config directory, the kas configuration,
+and the recipes. The config files of the layer are responsible for setting up
+the platform and distro configuration. The `conf/layer.conf` passes on the
+information where to find the recipes. The kas configuration file contains
+details, how to create the build and. It also inform about repositories that the
+image is based on. 
+
+
+To create the build, enter the command:
+```
+SHELL=bash kas-docker --ssh-dir ~/ssh-keys build meta-pcengines/kas.yml
+```
+
+Building the dom0 image can take several hours. At the end of the build, you
+should see a similar log:
+
+```
+NOTE: Executing Tasks
+NOTE: Setscene tasks completed
+NOTE: Tasks Summary: Attempted 2871 tasks of which 0 didn't need to be rerun and all
+ succeeded.
+```
+
+At that point you have to flash a bootable drive. On this spot  I will use the
+dd command. Check a directory of the drive to be flashed:
+
+```
+sudo fdisk -l
+```
+
+Change the directory to `<build-dir>/build/tmp/deploy/images/pcengines-apu2` and
+flash the drive:
+
+```
+sudo dd if=xen-dom0-image-pcengines-apu2.hddimg of=/dev/<drive_dir>
+```
+
+If everything went well, you should see the bootlog:
+
+[![asciicast](https://asciinema.org/a/mzfBdiDWNiWPOEEkev2F6CxwX.svg)](https://asciinema.org/a/mzfBdiDWNiWPOEEkev2F6CxwX?t=95)
+
+# VM guest configuration
+
+Once you have created the dom0 on your APU platform, it is time to place a VM
```

Comment:
Once you use `apu2`, now `APU`. To be fair, I'm not sure which should be used. Let's verify and be consistent about that.

Reply:
I think I should stick to apu2, as it is referred in the [docs](https://pcengines.ch/apu2.htm).